plugins/inputs/tail/tail.go

```diff
@@ -86,89 +91,88 @@ func (t *Tail) Start(acc telegraf.Accumulator) error {
 	defer t.Unlock()
 
 	t.acc = acc
-	t.tailers = make(map[string]*tail.Tail)
-
-	return t.tailNewFiles(t.FromBeginning)
-}
-
-func (t *Tail) tailNewFiles(fromBeginning bool) error {
-	var seek *tail.SeekInfo
-	if !t.Pipe && !fromBeginning {
-		seek = &tail.SeekInfo{
-			Whence: 2,
-			Offset: 0,
-		}
-	}
+	t.lines = make(chan *logline.LogLine)
+	t.files = make(map[string]bool)
 
 	var poll bool
 	if t.WatchMethod == "poll" {
 		poll = true
 	}
 
-	// Create a "tailer" for each file
+	var err error
+	t.watcher, err = watcher.NewLogWatcher(t.PollInterval.Duration, !poll, watcher.Logger(taillog.DiscardingLogger))
+	if err != nil {
+		defer close(t.lines)
```

Comment:
I'd skip defer here and elsewhere in this function: ordering is more confusing, it is more expensive, and we are going to return anyway.